eternalcore-core/src/main/java/com/eternalcode/core/feature/privatechat/toggle/PrivateChatToggleCommand.java

```diff
@@ -0,0 +1,125 @@
+package com.eternalcode.core.feature.privatechat.toggle;
+
+import com.eternalcode.annotations.scan.command.DescriptionDocs;
+import com.eternalcode.core.injector.annotations.Inject;
+import com.eternalcode.core.notice.NoticeService;
+import dev.rollczi.litecommands.annotations.argument.Arg;
+import dev.rollczi.litecommands.annotations.command.Command;
+import dev.rollczi.litecommands.annotations.context.Context;
+import dev.rollczi.litecommands.annotations.execute.Execute;
+import dev.rollczi.litecommands.annotations.optional.OptionalArg;
+import dev.rollczi.litecommands.annotations.permission.Permission;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+@Command(name = "msgtoggle")
+@Permission("eternalcore.msgtoggle")
+public class PrivateChatToggleCommand {
+
+    private final PrivateChatToggleService privateChatToggleService;
+    private final NoticeService noticeService;
+
+    @Inject
+    public PrivateChatToggleCommand(PrivateChatToggleService privateChatToggleService, NoticeService noticeService) {
+        this.privateChatToggleService = privateChatToggleService;
+        this.noticeService = noticeService;
+    }
+
+    @Execute
+    @DescriptionDocs(description = "Toggle receiving private messages on and off")
+    public void execute(@Context Player context, @OptionalArg PrivateChatToggleState state) {
+        UUID uniqueId = context.getUniqueId();
+
+        if (state == null) {
+            CompletableFuture<PrivateChatToggleState> privateChatToggleState = this.privateChatToggleService.getPrivateChatToggleState(context.getUniqueId());
+
+            privateChatToggleState.thenAccept(toggleState -> {
+                if (toggleState.equals(PrivateChatToggleState.OFF)) {
```

Comment:
You don't need to compare objects here; reference comparison is sufficient.
```suggestion
                if (toggleState == PrivateChatToggleState.OFF) {
```